ansible/roles/service/hippo_authoring/files/dev/shipit

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+function main {
+	local source_folder=$1
+	local destination_folder=$2
+
+	if [ -z "${source_folder}" ]; then
+		error "please specify source_folder"
+	fi
+
+	if [ -z "${destination_folder}" ]; then
+		error "please specify destination_folder"
+	fi
+
+	shipit "${source_folder}" "${destination_folder}"
+}
+
+function shipit {
+	local source_folder=$1
+	local destination_folder=$2
+	local apps_running=
+	local apps_status=
+
+	source ~/.hippo-manager-env
+	local tomcat_manager_url="http://${TOMCAT_MANAGER_USER}:${TOMCAT_MANAGER_PASSWORD}@${TOMCAT_MANAGER_HOST}/manager/text"
+
+	echo "Stop Hippo...                $(date +%T)"
+	sudo /bin/systemctl stop hippo_authoring
+
+	echo "Ship Hippo...                $(date +%T)"
+	find ${source_folder} -name "*.zip" | head -n 1 \
+		| xargs -I{} unzip -oq {} 'webapps/*' 'shared/*' -d "${destination_folder}"
+
+	echo "Start Hippo...               $(date +%T)"
+	sudo /bin/systemctl start hippo_authoring
+
+	echo "Wait for Hippo to wake up    $(date +%T)"
+	apps_status=$(curl -s -m4 ${tomcat_manager_url}/list | grep -E "^/(cms|site)")
```

Comment:
Worth using something like <https://stackoverflow.com/questions/16489809/emulating-a-do-while-loop-in-bash> to remove duplication?